lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -972,14 +974,23 @@ computeUtxoStatistics btype utxos =
 
 -- | A slot identifier is the combination of an epoch and slot.
 data SlotId = SlotId
-  { epochNumber :: !Word64
-  , slotNumber :: !Word16
+  { epochNumber :: !EpochNo
+  , slotNumber :: !SlotNo
   } deriving stock (Show, Read, Eq, Ord, Generic)
 
+newtype SlotNo = SlotNo { unSlotNo :: Word16}
+    deriving stock (Show, Read, Eq, Ord, Generic)
+    deriving newtype (Num, Buildable, NFData, Enum)
+
+newtype EpochNo = EpochNo { unEpochNo :: Word64 }
+    deriving stock (Show, Read, Eq, Ord, Generic)
+    deriving newtype (Num, Buildable, NFData, Enum)
```

Comment:
I added a `Num` instance here.

I didn't end up adding a `Integral` instance, which arguably could have saved a few `unEpochNo`/`unSlotNo`, because I don't feel completely confident about it being a good idea.

Reply:
> I didn't end up adding a Integral instance, which arguably could have saved a few unEpochNo/unSlotNo, because I don't feel completely confident about it being a good idea.

And I second this (for not being a good idea) :)